### PR TITLE
Feature/update viz setting and widgets

### PR DESCRIPTION
**Pull request by sudan45**
Addresses:
 - https://github.com/the-deep/server/pull/1393

This PR includes changes from
- https://github.com/the-deep/server/pull/1425
- https://github.com/the-deep/server/pull/1412

## This PR doesn't introduce any:

- [X] temporary files, auto-generated files or secret keys
- [X] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [ ] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/analysis_framework/migrations/0041_widget_mapping.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            []
+        )
+
+        # -- Organigram
+        af_prop['stats_config']['organigram_widgets'] = (
+            af_prop['stats_config'].get('affected_groups_widgets') or
+            af_prop['stats_config'].get('organigram_widgets') or
+            af_prop['stats_config'].get('organigram_widget') or
+            []
+        )
```
Comment (thenav56) [inline apps/analysis_framework/migrations/0041_widget_mapping.py]: From existing code, we had affected_groups_widget, which should be [...get('affected_groups_widgets')]

Do we have affected_groups_widgets as well?
Do we have organigram_widget as well? It should be array right?

File: apps/analysis_framework/migrations/0041_widget_mapping.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        # -- Multiselect
+        af_prop['stats_config']['multiselect_widgets'] = (
+            af_prop['stats_config'].get('specific_needs_groups_widgets') or
+            af_prop['stats_config'].get('multiselect_widgets') or
+            af_prop['stats_config'].get('multiselect_widget') or
+            af_prop['stats_config'].get('demographic_group_widget') or
+            []
+        )
```
Comment (thenav56) [inline apps/analysis_framework/migrations/0041_widget_mapping.py]: Do we have specific_needs_groups_widgets?

For multiselect_widget, demographic_group_widget they should be wrapped with an array

We are also missing demographic_groups_widgets here

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good. We need more check for the migration.

File: apps/analysis_framework/migrations/0041_widget_mapping.py (hunk 1 above)
Comment (sudan45) [reply apps/analysis_framework/migrations/0041_widget_mapping.py]: @thenav56  dai we do have organigram widget in alpha 1 server

File: apps/analysis_framework/migrations/0041_widget_mapping.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            af_prop['stats_config'].get('widget1d') or
+            []
+        )
+
+        # -- Widget2D
+        af_prop['stats_config']['widget_2d'] = (
+            af_prop['stats_config'].get('widget_2d') or
+            af_prop['stats_config'].get('widget2d') or
+            []
+        )
```
Comment (thenav56) [inline apps/analysis_framework/migrations/0041_widget_mapping.py]: We need to use the old one as fallback
```suggestion
        af_prop['stats_config']['widget_2d'] = (
            af_prop['stats_config'].get('widget2d') or
            af_prop['stats_config'].get('widget_2d') or
            []
        )
```

**Review by thenav56 — APPROVED** [review]

**Comment by sudan45** [comment]
@AdityaKhatri dai and @subinasr  lets test together on alpha 1 server

**Pull request merged**